src/WinRT.Runtime/Marshalers.cs

```diff
@@ -1570,7 +1570,11 @@ private static IObjectReference CreateMarshalerCore(T value)
 #else
     public
 #endif
-    static class MarshalInspectable<T>
+    static class MarshalInspectable<

```

Comment:
I really, _really_ would like to avoid changing this back, and same for `ComWrappersSupport`. It's not so much for the size regression here (though I do want to run sizoscope later as well), but also because this attribute is viral and it will have to be propagated everywhere. This means that every single code path ultimately using any of these APIs will also need to have the attribute applied to it, which is extremely clunky and annoying, and something that all library authors and app developers using these CsWinRT APIs in some way would have to do. It also means that you'd be unnecessarily rooting non public constructors for all managed types in authoring scenarios where you want to get a CCW for them (via `FromManaged`), or you'd have to manually suppress those yourself, which again is extremely clunky and not intuitive at all. For instance, I hit this last scenario in multiple places in ComputeSharp, in my Win2D interop library.

Can we find some other way to work around this that doesn't require us to revert these changes? If this is a niche enough scenario, can we perhaps just use documentation to tell people that hit this (if any) to manually root things via the .rd.xml file, or something else? Or, could we maybe come up with some other fallback path with `[DynamicDependency]` or something, that we could do only when not on NAOT?

Basically anything but bring these annotations back 😭

Reply:
Let me see if maybe it is possible to reduce the number of attributes we need maybe?  DynamicDependency probably won't work because this can be referencing any projection assembly.  This is used during a core scenario (getting an existing object from native code), so would be hard to ask folks to workaround it.

Reply:
Another idea could be to have the generator emit the `[DynamicDependency]` attributes on some method (say, a module initializer, or a partial declaration of something or whatever) after identifying what types are needed? At least it would be pay for play. We could also maybe introduce some MSBuild property for devs to indicate which outdated projections they're using, and we could emit those annotations only for types from there? Basically I really want to avoid adding the annotations for everyone. If the issue is that one backcompat scenario, let's try to find some workaround specifically for that without making the dev experience worse for everyone 🥲